modules/call-dispatch/src/lib.rs

```diff
@@ -291,6 +298,18 @@ impl<T: Config<I>, I: Instance> MessageDispatch<T::MessageId> for Module<T, I> {
 			return;
 		}
 
+		// filter the call
+		if !T::CallFilter::filter(&call) {
```

Comment:
I wonder if it wouldn't be better to move the filtering before weight checks. I'm expecting that the filtering might be much less complicated than weight calculations and it it has a nice consequence of preventing exploiting incorrectly calculated weights (i.e. you generate excessive load by forcing the chain to calculate it even though the call is filtered out).